services/rest/websocket/handler.go

```diff
@@ -135,15 +155,31 @@ func writeSuccessResponse(conn connection.Conn, serialize serialization.Serializ
 	conn.WriteMessage(messageType, success)
 }
 
-func writeBadRequestResponse(conn connection.Conn, serialize serialization.SerializeFunc, messageType int, err error) {
+func writeBadRequestResponse(conn connection.Conn, serialize serialization.SerializeFunc, messageType int, reqGuid string, err error) {
 	response := &pb.SendEventResponse{
 		Status:   pb.Status_STATUS_ERROR,
 		Code:     pb.Code_CODE_BAD_REQUEST,
 		SentTime: time.Now().Unix(),
 		Reason:   fmt.Sprintf("cannot deserialize request: %s", err),
-		Data:     nil,
+		Data: map[string]string{
```

Comment:
(suggestion): If the name `Data` could be changed, I would suggest for `Properties` as this appears to be a `property bag` with a `(key/val)` to me.

Reply:
I actually feel it the other way, `Data` seems to be very generic and can be used to send anything in the future. whereas the `Properties` not so much, like property of what? Also, this will be a change in the contracts, and SDK level changes might be required. So we can't change it for now.

Reply:
but, in this case, `data` refers to a key/value pair. Any data can be sent, but it must always be in the shape of a map.
Okay, for now we can leave it as is because the contract change is necessary.